centaur/src/main/scala/centaur/serialization/TestsReports.scala

```diff
@@ -0,0 +1,76 @@
+package centaur.serialization
+
+import java.io.{BufferedWriter, FileNotFoundException, FileWriter}
+
+import org.slf4j.LoggerFactory
+
+import scala.collection.concurrent.TrieMap
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+/**
+  * This class contains information about last version in which corresponding integration test was successfully run
+  *
+  * @param testCasesRuns key is a test name; value is a version of Cromwell, in which this test passed last time
+  */
+case class TestsReports(testCasesRuns: TrieMap[String, String]) {
+  def addSuccessfulTest(testName: String, version: String): Option[String] =
+    testCasesRuns.put(testName, version)
```

Comment:
There's something bothers me about this class:
1. `testCasesRuns: TrieMap[String, String]` is somewhat obscure, without comment it's hard to say what `String` key is and what is another `String` value. I suggest to move it to separate type like:
```scala
type TestName = String
type CromwellVersion = String
TrieMap[TestName, CromwellVersion]
```
2. Do we need this case class at all? I see as @iliaonishchenko said you already serializing `TrieMap` and `addSuccessfulTest` doesn't add anything new to  the `put` method.

Reply:
1. Agreed.
2. I think we need some context for this `TrieMap`, so wrapping it in a case class and encapsulating some methods is just a good practice. Let's discuss this. @iliaonishchenko what do you think about this?

Reply:
We can go even further and create type for `TrieMap[TestName, CromwellVersion]`. For example `TestReports`. In that case it will be crystal clear with the case class.

The most important question is would we like to give users access to all methods of TrieMap. If yes, then we can remove case class.

Reply:
> We can go even further and create type for `TrieMap[TestName, CromwellVersion]`. For example `TestReports`. In that case it will be crystal clear with the case class.

Good idea!

> The most important question is would we like to give users access to all methods of TrieMap. If yes, then we can remove case class.

I don't see why not. Anyway with case class you have all access to this map and if you really want to encapsulate then we need to make it private and it already starts looking like Java.